cmd/habitat/community/manager.go

```diff
@@ -81,57 +126,81 @@ func (m *Manager) CreateCommunity(name string, createIpfs bool) (*community.Comm
 		return nil, err
 	}
 
-	err = m.clusterManager.CreateCluster(communityID)
+	initState := &community.CommunityState{
+		CommunityID: communityID,
+	}
+	stateBuf, err := json.Marshal(initState)
+	if err != nil {
+		return nil, err
+	}
+
+	updateChan, err := m.clusterManager.CreateCluster(communityID, stateBuf)
+	if err != nil {
+		return nil, err
+	}
+
+	stateMachine, err := state.NewCommunityStateMachine(initState, updateChan, &ClusterDispatcher{
+		communityID:    communityID,
+		clusterManager: m.clusterManager,
+	}, NewCommunityExecutor(&m.processManager))
+	if err != nil {
+		return nil, err
+	}
+
+	err = m.addCommunity(communityID, stateMachine)
 	if err != nil {
 		return nil, err
 	}
 
-	// TODO @eagraf have this be downstream of a Raft update
 	if createIpfs {
-		err, swarmkey, peerid, addrs := m.config.NewCommunityIPFSNode(name, filepath.Join(m.Path, communityID))
+		// TODO @eagraf find a better way to guarantee this runs after the first election
```

Comment:
seems like we need some sort of signal wrapping around all processes needed by habitat - or to start ipfs with the process manager instead of treating it separately. hard though because it is such a big component of habitat, which basically doesn't run without it :/

Reply:
This gets changed in a subsequent commit I think